### ISSUE TITLE
Control-C issues duplicate signals to container

### ISSUE THREAD

**Issue by ryanuber** [docker/for-mac#5244]
- [ ] I have tried with the latest version of Docker Desktop
  - [ ] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID: BD253A61-8FD5-49F8-BC42-41EFE6A5AC2A/20210112212415

### Expected behavior

Running a container in the foreground via `docker run` and pressing ctrl-c sends exactly one signal (defined by STOPSIGNAL in the Dockerfile) to the running process.

### Actual behavior

Two signals are sent to the running container, causing undesired behavior in the running application. Running with `-it` (interactive, TTY) produces a single signal, as expected.

### Information
  - Is it reproducible? Yes
  - Is the problem new? Yes - this seems to have been a behavior change at Docker Desktop 2.4.0. Prior versions send a single signal as expected.
  - Did the problem appear with an update? Versions from 2.4.0 onward exhibit the unexpected multiple signals behavior.
  - macOS Version: 10.15.7

### Steps to reproduce the behavior

1. Create a `Dockerfile` with the following content:
```Dockerfile
FROM ubuntu:20.04

COPY sig.sh /sig.sh
RUN chmod +x /sig.sh

STOPSIGNAL SIGINT

ENTRYPOINT ["/sig.sh"]
``` 
2. Create a script file `sig.sh` with the following content:
```bash
#!/bin/bash

trap "register" INT

register() {
    echo "INT"
    [ -z "$SIG" ] && SIG=1 || exit
}

while true; do
    sleep 100 &
    wait
done
```
3. Build the container with `docker build -t signal-test:latest .`
4. Run the container with `docker run signal-test:latest`
5. Press ctrl+c once. Note that `INT` is printed twice, thus indicating that two signals were sent to the process.

**Comment by stephen-turner** [comment]
Interesting. I can reproduce this on Windows too.

**Comment by StefanScherer** [comment]
Indeed, there is a different behaviour between a Linux-only environment and Docker Desktop.

I've  built your test image and ran it on **Ubuntu 20.04** with Docker Engine 20.10.2, and I have to press CTRL+C twice to exit the container:

![signal-test-on-ubuntu](https://user-images.githubusercontent.com/207759/104761538-58e95e80-5763-11eb-96f7-cb71f25ce509.gif)

Docker Desktop for Mac 3.1.0 with Docker Engine 20.10.2 sends the signal twice with just one CTRL+C:

![signal-test-on-mac](https://user-images.githubusercontent.com/207759/104761559-60106c80-5763-11eb-9380-68da28a759ca.gif)

**Comment by thaJeztah** [comment]
does it still reproduce if you disable "cloud experience" under experimental features in the Docker Desktop preferences?

**Comment by StefanScherer** [comment]
@thaJeztah Good catch!

Without the cloud experience (or running `com.docker.cli` directly) it behaves just like directly on a Linux system:

![signal-test-on-mac-with-com docker cli](https://user-images.githubusercontent.com/207759/104764266-6ef91e00-5767-11eb-84cc-120d95a9a16e.gif)

**Comment by ryanuber** [comment]
Interesting! I had only disabled the CLI experimental features, but I confirmed disabling Cloud Experience does indeed fix the problem. `com.docker.cli` executed directly while Cloud Experience is enabled also works as expected. Nice catch!

**Comment by thaJeztah** [comment]
@rumpl @gtardif ^ ptal looks to be an issue in https://github.com/docker/compose-cli
